internal/cli/command/controlplane/debug.go

```diff
@@ -0,0 +1,241 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplane
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"emperror.dev/errors"
+	"github.com/banzaicloud/banzai-cli/internal/cli"
+	"github.com/shirou/gopsutil/host"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+)
+
+const basePath = "pipeline-debug-bundle/"
+
+type debugOptions struct {
+	filename string
+	*cpContext
+}
+
+func (d *debugOptions) filepath() string {
+	if !d.cpContext.flags.Changed("filename") {
+		d.filename = fmt.Sprintf("pipeline-debug-bundle-%s.tgz", time.Now().Format("20060102-1504"))
+	}
+	return filepath.Join(d.cpContext.workspace, d.filename)
+}
+
+// NewDebugCommand creates a new cobra.Command for `banzai pipeline debug`.
+func NewDebugCommand(banzaiCli cli.Cli) *cobra.Command {
+	options := debugOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "debug",
+		Short: "Generate debug bundle",
+		Long:  "Collect and package status information and configuration data that is needed by Banzai Cloud support team to resolve issues on a remote Pipeline installation",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceErrors = true
+			cmd.SilenceUsage = true
+
+			return runDebug(options, banzaiCli)
+		},
+	}
+
+	options.cpContext = NewContext(cmd, banzaiCli)
+	flags := cmd.Flags()
+	flags.StringVarP(&options.filename, "filename", "f", "", "Name or path to output file relative to the workspace (prefix with ./ for current working directory; default: pipeline-debug-bundle-DATE.tgz)")
```

Comment:
This is indeed a different context, and does not name an existing file unlike in case of inputs, but controls the naming of the result. `-f`/`--output-file` might be better?

Reply:
I think `--output-file` would be much better if we want to distinguish outputs from inputs - otherwise `--file` is the consistent choice.

Reply:
I would even consider `-o` for shorthand, but I leave it up to you.

Reply:
I'll go with `-O`, becuse o is in use globally (for output format).